upup/pkg/api/cluster.go

```diff
@@ -240,11 +240,13 @@ type EtcdClusterSpec struct {
 
 type EtcdMemberSpec struct {
 	// Name is the name of the member within the etcd cluster
-	Name string `json:"name,omitempty"`
-	Zone string `json:"zone,omitempty"`
+	Name            string `json:"name,omitempty"`
+	Zone            string `json:"zone,omitempty"`
 
-	VolumeType string `json:"volumeType,omitempty"`
-	VolumeSize int    `json:"volumeSize,omitempty"`
+	VolumeType      string `json:"volumeType,omitempty"`
+	VolumeSize      int    `json:"volumeSize,omitempty"`
+	KmsKeyId        string `json:"kmsKeyId,omitempty"`
+	EncryptedVolume bool   `json:"encryptedVolume,omitempty"`
```

Comment:
ALL new fields that are optional (which should be all fields) must be
pointers in their Go types.  ALL.  EVERY ONE.  Yes I know exactly how
tedious that is.

On Wed, Sep 7, 2016 at 9:53 AM, Justin Santa Barbara <
notifications@github.com> wrote:

> In upup/pkg/api/cluster.go
> <https://github.com/kubernetes/kops/pull/398#discussion_r77862147>:
>
> >
> > -	VolumeType string `json:"volumeType,omitempty"`
> > -	VolumeSize int    `json:"volumeSize,omitempty"`
> > +	VolumeType      string `json:"volumeType,omitempty"`
> > +	VolumeSize      int    `json:"volumeSize,omitempty"`
> > +	KmsKeyId        string `json:"kmsKeyId,omitempty"`
> > +	EncryptedVolume bool   `json:"encryptedVolume,omitempty"`
>
> I think this should be *bool, so we can tell the difference between false
> and not-set.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kops/pull/398/files/8c3b2274d91e1c7e3994eeb4cc5450f17fd966d4#r77862147>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFVgVIWRLDNjxicsqJW7Ol3k06FEqZ-iks5qnuwfgaJpZM4JzDC_>
> .
>